python/cugraph/structure/shuffle.py

```diff
@@ -0,0 +1,46 @@
+import math
+from dask.dataframe.shuffle import rearrange_by_column
+
+
+def get_n_workers():
+    from dask.distributed import default_client
+    client = default_client()
+    return len(client.scheduler_info()['workers'])
+
+
+def get_2D_div(ngpus):
+    pcols = int(math.sqrt(ngpus))
+    while ngpus % pcols != 0:
+        pcols = pcols - 1
+    return int(ngpus/pcols), pcols
+
+
+def _set_partitions_pre(df, num_verts, prows, pcols):
+    rows_per_div = math.ceil(num_verts/prows)
+    cols_per_div = math.ceil(num_verts/pcols)
+    partitions = df['src'].floordiv(rows_per_div) * pcols + df['dst'].floordiv(cols_per_div)
+    return partitions
+
+
+def shuffle(dg):
+    ddf = dg.edgelist.edgelist_df
+    ngpus = get_n_workers()
+    prows, pcols = get_2D_div(ngpus)
```

Comment:
Should probably add a check that prows * pcols = ngpus.  The implementation of get_2D_div guarantees that, but if the user specifies a value to tune things that isn't guaranteed.

Reply:
added